src/main/java/com/aws/greengrass/mqttclient/IotCoreTopicValidator.java

```diff
@@ -148,7 +147,8 @@ private static void validateEffectiveTopic(String effectiveTopic, Operation oper
     private static Optional<String> removePrefix(String topic, String prefixRegex) {
         String[] firstAndRest = topic.split(prefixRegex, 2);
         if (firstAndRest.length == 2) {
-            return Optional.of(firstAndRest[1]);
+            // intern to deduplicate topic in memory
+            return Optional.of(firstAndRest[1].intern());
```

Comment:
this isn't needed. The string is already created, so it is too late. Interning is actually going to spend more time and won't really save memory

Reply:
ah okay, fixed